api/v1alpha1/kongplugin_installation.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2022 Kong Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&KongPluginInstallation{}, &KongPluginInstallationList{})
+}
+
+//+genclient
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:resource:shortName=kpi,categories=kong;all
+// +kubebuilder:subresource:status
+//+kubebuilder:printcolumn:name="Ready",description="The Resource is ready",type=string,JSONPath=`.status.conditions[?(@.type=='Ready')].status`
+
+// KongPluginInstallation is the Schema
+type KongPluginInstallation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   KongPluginInstallationSpec   `json:"spec,omitempty"`
+	Status KongPluginInstallationStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// KongPluginInstallationList contains a list of KongPluginInstallation.
+type KongPluginInstallationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []KongPluginInstallation `json:"items"`
+}
+
+// KongPluginInstallationSpec defines the desired state of KongPluginInstallation.
+type KongPluginInstallationSpec struct {
+
+	// Image is OCI image reference for the Kong Plugin.
+	//
+	// +kubebuilder:validation:Required
+	Image string `json:"image"`
+
+	// SecretRef allows specifying secret to be used for OCI registry authentication to pull the image.
+	// When omitted it is assumed that the image is public.
+	//
+	// +optional
+	SecretRef *corev1.SecretReference `json:"secretRef,omitempty"`
+}
+
+// KongPluginInstallationStatus defines the observed state of KongPluginInstallation.
+type KongPluginInstallationStatus struct {
+	// Conditions describe the current conditions of the KongPluginInstallation.
+	//
+	// Known condition types are:
+	//
+	//   - "Accepted"
+	//   - "Fetching"
+	//   - "PluginFetched"
```

Comment:
- IMHO it'd be better (as suggested above) to make those part of the API and propose them in this PR. This would also allow adding comments to condition type and reason consts to let users know when they can expect them.
- `Fetching` doesn't seem like a good candidate for a condition `Type`. Perhaps this was meant to be condition's reason?